store/src/java/com/zimbra/cs/service/admin/Auth.java

```diff
@@ -210,121 +208,119 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         return doResponse(request, at, zsc, context, acct, csrfSupport);
     }
 
+    private void doCSRFCheck(Map<String, Object> context) throws ServiceException {
+
+        ZimbraLog.extensions.debug("doCSRFCheck in");
+        HttpServletRequest req = (HttpServletRequest) context.get(SoapServlet.SERVLET_REQUEST);
+        HttpServletResponse res = (HttpServletResponse) context.get(SoapServlet.SERVLET_RESPONSE);
+        AuthToken authToken = null;
+        try {
+            authToken = AuthUtil.getAuthTokenFromHttpReq(req, res, true, false);
+            ZimbraLog.extensions.debug("doCSRFCheck authToken %s", authToken);
+        } catch (IOException e) {
+            ZimbraLog.security.warn("Problem accessing auth token:", e);
+        }
+
+        String csrfToken = null;
+        Cookie[] cookies = req.getCookies();
+        if (cookies != null) {
+            for (Cookie cookie : cookies) {
+                if (authToken == null) {
+                    if (cookie.getName().equals("ZM_ADMIN_AUTH_TOKEN")) {
+                        authToken = AuthUtil.getAdminAuthTokenFromCookie(req);
+                    }
+                }
+                if (cookie.getName().equals(Constants.CSRF_TOKEN)) {
```

Comment:
Let's use if else block or something better, because a given cookie at an iteration can result either in AuthToken or CSRFToken, but not both
Try this:
```
for (Cookie cookie : cookies) {
	if (cookie.getName().equals("ZM_ADMIN_AUTH_TOKEN") && authToken == null) {
		authToken = AuthUtil.getAdminAuthTokenFromCookie(req);
	} else if (cookie.getName().equals(Constants.CSRF_TOKEN)) {
		csrfToken = cookie.getValue();
	}
}
```

Reply:
Done